### PR TITLE
Fixes for several issues in JenkinsQueueJob task

### PR DESCRIPTION
**Pull request by kelliejos**
Fixes for #7527, a bug in how tree query parameters are used, and for an issue where builds are failing when the network connection is lost while streaming the Jenkins console.

Tested with local deployment and ran L0 tests

### CODE REVIEW

File: Tasks/JenkinsQueueJobV2/job.ts [hunk 1]
```diff
@@ -276,7 +276,9 @@ export class Job {
                 if (thisJob.queue.TaskOptions.capturePipeline) {
                     const downstreamProjects = thisJob.Search.ParsedTaskBody.downstreamProjects || [];
                     downstreamProjects.forEach((project) => {
-                        new Job(thisJob.queue, thisJob, project.url, null, -1, project.name); // will add a new child to the tree
+                        if (project.color !== 'disabled') {
```
Comment (kelliejos) [inline Tasks/JenkinsQueueJobV2/job.ts]: The 'color' field is how Jenkins identifies the status of a project. This field will be set to disabled if the project has been disabled.

File: Tasks/JenkinsQueueJobV2/job.ts [hunk 2]
```diff
@@ -445,7 +447,7 @@ export class Job {
                     thisJob.stopWork(0, JobState.Finishing);
                 }
             }
-        }).auth(thisJob.queue.TaskOptions.username, thisJob.queue.TaskOptions.password, true)
+        }).auth(thisJob.queue.TaskOptions.username, thisJob.queue.TaskOptions.password, false)
```
Comment (kelliejos) [inline Tasks/JenkinsQueueJobV2/job.ts]: This parameter is the 'sendImmeadiately' flag. When set to false it forces the request to retry with the proper authentication headers. See this doc for reference: https://www.npmjs.com/package/request#http-authentication)

File: Tasks/JenkinsQueueJobV2/jobsearch.ts [hunk 3]
```diff
@@ -242,7 +242,7 @@ export class JobSearch {
             thisSearch.stopWork(0); // found everything we were looking for
             return;
         } else {
-            const url: string  = util.addUrlSegment(thisSearch.taskUrl, thisSearch.nextSearchBuildNumber + '/api/json?tree=actions,timestamp');
+            const url: string  = util.addUrlSegment(thisSearch.taskUrl, thisSearch.nextSearchBuildNumber + '/api/json?tree=actions[causes[shortDescription,upstreamBuild,upstreamProject,upstreamUrl]],timestamp');
```
Comment (kelliejos) [inline Tasks/JenkinsQueueJobV2/jobsearch.ts]: We should be specifying the fields we want returned in the actions, else we get the default empty actions block.

File: Tasks/JenkinsQueueJobV2/job.ts (hunk 2 above)
Comment (lkillgore) [reply Tasks/JenkinsQueueJobV2/job.ts]: Comment in code, plz!

Comment (kelliejos) [reply Tasks/JenkinsQueueJobV2/job.ts]: Done!

File: Tasks/JenkinsQueueJobV2/jobsearch.ts [hunk 4]
```diff
@@ -33,7 +33,7 @@ export class JobSearch {
     private foundCauses : any[] = []; // all found causes indexed by executableNumber
 
     public Initialized: boolean = false;
-    public ParsedTaskBody: {downstreamProjects?: {name: string, url: string}[], lastBuild?: {number: number}}; // the parsed task body of the job definition
+    public ParsedTaskBody: {downstreamProjects?: {name: string, url: string, color: string}[], lastBuild?: {number: number}}; // the parsed task body of the job definition
```
Comment (brcrista) [inline Tasks/JenkinsQueueJobV2/jobsearch.ts]: I would just define a named type for this -- would make it easier to read

File: Tasks/JenkinsQueueJobV2/job.ts (hunk 1 above)
Comment (brcrista) [reply Tasks/JenkinsQueueJobV2/job.ts]: So, we only add jobs to the tree if they are not disabled.  What about the root job -- can it be disabled?

Comment (kelliejos) [reply Tasks/JenkinsQueueJobV2/job.ts]: Attempting to queue a disabled root job results in a 404 from Jenkins, which causes our build to fail right away.

File: Tasks/JenkinsQueueJobV2/jobsearch.ts (hunk 4 above)
Comment (brcrista) [reply Tasks/JenkinsQueueJobV2/jobsearch.ts]:
```suggestion
interface Project {
    name: string,
    url: string,
    color: string
}

interface ParsedTaskBody {
    downstreamProjects?: Project[],
    lastBuild?: {
        number: number
    }
}

public ParsedTaskBody: ParsedTaskBody; // the parsed task body of the job definition
```

**Comment by go2guy** [comment]
Any idea on when this will be released and will it be version 3?

**Pull request merged**